cmd/skaffold/app/cmd/config/set.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func NewCmdSet(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set",
+		Short: "Set a value in the global skaffold config",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 2 {
+				return errors.New("Please provide field name and value to set")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			resolveKubectlContext()
+			return setConfigValue(args[0], args[1])
+		},
+	}
+	AddConfigFlags(cmd)
+	return cmd
+}
+
+func setConfigValue(name string, value interface{}) error {
```

Comment:
I don't think we want to have a reflection based config

Reply:
Could you explain your thinking here? Most config libraries (e.g. viper) are based on reflection anyway. Since the config structure here is actually more like multiple configs in one file (one for each kubectx we set values for), using reflection makes it easy to set values for multiple contexts at once without having to know anything about the values themselves.

Reply:
I don't think it's a requirement to set multiple contexts at the same time. Can we try to do this in an explicit way with just two levels? A yaml representation would be something like this in case of a default repo:  
````
global: 
  default-repo: gcr.io/myrepo1/
kubecontexts:
   - kubectx: minikube
         default_repo: gcr.io/myrepo2/
   - kubectx:  my_dev_gke
         default_repo: gcr.io/myrepo2/
````
so if I'm trying to use skaffold with the context "context3" it will just simply use the global definition of the default_repo.

In my head this should be easier to manage explicitly and no need for reflection but I might be missing something.

Reply:
I think we'll still need to use reflection to know which field we want to set in the config though. If I do `skaffold config set foo bar`, the only way to retrieve the `Foo` field from the config object (or even know that it exists in the first place) is to use reflection to get the field by name. I also might be missing something though :)

I agree that we don't necessarily have to set multiple context at once though, I could get rid of that. We could make it a requirement that only one kubectx can be passed (for example, if you did `skaffold config set foo bar -c context1 -c context2` we would error out). This would be consistent with having a "default" (or "global") context that we could treat as just another kubectx entry.

Your config structure would also work fine, but to me it's a bit more confusing to have a top level "config" (for `global`), and then another field with a bunch of nested configs inside of it. WDYT?

Reply:
re: reflection -  If @r2d4 has a better idea how to implement this, it would be great to hear! Reflection does feel heavy - I don't think we need to support a large amount of arbitrary values - what do you think about a simple switch-case statement for supported, single valued fields would be overly simple?  
re: global on top level vs same level - I'm okay with the same level approach. I do have 2 small thoughts to add for the record which if they become an issue we can probably handle relatively easily:
 - the separation can create a separate models for global vs context level configs, and thus we could have global only vars that are not overridable (not sure we need that yet, but there might be things that don't make sense on a context level) 
 - there can't be a kubecontext named "global" - but again this might not be a real issue.